sa/sa.go

```diff
@@ -1075,6 +1087,60 @@ func addFQDNSet(tx *gorp.Transaction, names []string, serial string, issued time
 	})
 }
 
+// addOrderFQDNSet creates a new OrderFQDNSet row using the provided
+// information. This function accepts a transaction so that the orderFqdnSet
+// addition can take place within the order addition transaction.
+func addOrderFQDNSet(
+	tx *gorp.Transaction,
+	names []string,
+	orderID int64,
+	regID int64,
+	expires time.Time) error {
+	if err := tx.Insert(&orderFQDNSet{
+		SetHash:        hashNames(names),
+		OrderID:        orderID,
+		RegistrationID: regID,
+		Expires:        expires,
+	}); err != nil {
+		return Rollback(tx, err)
```

Comment:
Hm... since the caller created the transaction I feel like it should be the one to roll it back, but I'm not 100% convinced if that is the right convention.

Reply:
I wasn't sure about this myself. Reading it again I think I agree that whoever created the TX should do the rollback. I'll change accordingly.

Reply:
Fixed in 6ce9f74